### ISSUE TITLE
Should not switch on kind when version is 4 in ReadCustomMetadata 

### ISSUE THREAD

**Issue by SimonCropp** [jbevain/cecil#189]
in ReadCustomMetadata https://github.com/jbevain/cecil/blob/master/symbols/pdb/Microsoft.Cci.Pdb/PdbFunction.cs#L346

I thinks this 

```
  if (version != 4) {
    throw new PdbDebugException("Unknown custom metadata item version: {0}", version);
  }
```

Is causing us problems https://github.com/Fody/Fody/issues/141#issuecomment-64018913

it seems to CCI version only does the switch on kind when the version is 4 https://ccimetadata.codeplex.com/SourceControl/latest#Sources/PdbReader/PdbFunction.cs

```
private void ReadCustomMetadata(BitAccess bits) {
  int savedPosition = bits.Position;
  byte version;
  bits.ReadUInt8(out version);
  byte kind;
  bits.ReadUInt8(out kind);
  bits.Align(4);
  uint numberOfBytesInItem;
  bits.ReadUInt32(out numberOfBytesInItem);
  if (version == 4) {
    switch (kind) {
      case 0: this.ReadUsingInfo(bits); break;
      case 1: this.ReadForwardInfo(bits); break;
      case 2: break; // this.ReadForwardedToModuleInfo(bits); break;
      case 3: this.ReadIteratorLocals(bits); break;
      case 4: this.ReadForwardIterator(bits); break;
    }
  }
  bits.Position = savedPosition + (int)numberOfBytesInItem;
}
```

This was done on Nov11 as part of this https://ccimetadata.codeplex.com/SourceControl/changeset/73073

Want me to attempt a pull request?

**Comment by bradphelan** [comment]
I just got this error trying to port my app to VS 2015 / Rosyln

```
All packages are already installed and there is nothing to restore.
NuGet package restore finished.
1>------ Build started: Project: Numerics (Numerics\Numerics), Configuration: Release Any CPU ------
1>  Fody (version 1.13.0.0) Executing
1>MSBUILD : error : Fody: An unhandled exception occurred:\r
1>MSBUILD : error : Exception:\r
1>MSBUILD : error : Unknown custom metadata item kind: 7\r
1>MSBUILD : error : StackTrace:\r
1>MSBUILD : error :    at Microsoft.Cci.Pdb.PdbFunction.ReadCustomMetadata(BitAcc
1>ess bits) in c:\Code\cecil\symbols\pdb\Microsoft.Cci.Pdb\PdbFunction.cs:line 35
1>9\r
1>MSBUILD : error :    at Microsoft.Cci.Pdb.PdbFunction..ctor(ManProcSym proc, Bi
1>tAccess bits) in c:\Code\cecil\symbols\pdb\Microsoft.Cci.Pdb\PdbFunction.cs:lin
1>e 270\r
1>MSBUILD : error :    at Microsoft.Cci.Pdb.PdbFunction.LoadManagedFunctions(BitA
1>ccess bits, UInt32 limit, Boolean readStrings) in c:\Code\cecil\symbols\pdb\Mic
1>rosoft.Cci.Pdb\PdbFunction.cs:line 136\r
1>MSBUILD : error :    at Microsoft.Cci.Pdb.PdbFile.LoadFuncsFromDbiModule(BitAcc
1>ess bits, DbiModuleInfo info, IntHashTable names, ArrayList funcList, Boolean r
1>eadStrings, MsfDirectory dir, Dictionary`2 nameIndex, PdbReader reader) in c:\C
1>ode\cecil\symbols\pdb\Microsoft.Cci.Pdb\PdbFile.cs:line 273\r
1>MSBUILD : error :    at Microsoft.Cci.Pdb.PdbFile.LoadFunctions(Stream read, Di
1>ctionary`2& tokenToSourceMapping, String& sourceServerData, Int32& age, Guid& g
1>uid) in c:\Code\cecil\symbols\pdb\Microsoft.Cci.Pdb\PdbFile.cs:line 382\r
1>MSBUILD : error :    at Mono.Cecil.Pdb.PdbReader.PopulateFunctions() in c:\Code
1>\cecil\symbols\pdb\Mono.Cecil.Pdb\PdbReader.cs:line 94\r
1>MSBUILD : error :    at Mono.Cecil.Pdb.PdbReader.ProcessDebugHeader(ImageDebugD
1>irectory directory, Byte[] header) in c:\Code\cecil\symbols\pdb\Mono.Cecil.Pdb\
1>PdbReader.cs:line 75\r
1>MSBUILD : error :    at Mono.Cecil.ModuleDefinition.ProcessDebugHeader() in c:\
1>Code\cecil\Mono.Cecil\ModuleDefinition.cs:line 786\r
1>MSBUILD : error :    at Mono.Cecil.ModuleDefinition.ReadSymbols(ISymbolReader r
1>eader) in c:\Code\cecil\Mono.Cecil\ModuleDefinition.cs:line 859\r
1>MSBUILD : error :    at Mono.Cecil.ModuleReader.ReadSymbols(ModuleDefinition mo
1>dule, ReaderParameters parameters) in c:\Code\cecil\Mono.Cecil\AssemblyReader.c
1>s:line 108\r
1>MSBUILD : error :    at Mono.Cecil.ModuleReader.CreateModuleFrom(Image image, R
1>eaderParameters parameters) in c:\Code\cecil\Mono.Cecil\AssemblyReader.cs:line 
1>83\r
1>MSBUILD : error :    at Mono.Cecil.ModuleDefinition.ReadModule(Stream stream, R
1>eaderParameters parameters) in c:\Code\cecil\Mono.Cecil\ModuleDefinition.cs:lin
1>e 890\r
1>MSBUILD : error :    at Mono.Cecil.ModuleDefinition.ReadModule(String fileName,
1> ReaderParameters parameters) in c:\Code\cecil\Mono.Cecil\ModuleDefinition.cs:l
1>ine 874\r
1>MSBUILD : error :    at InnerWeaver.ReadModule() in c:\TeamCity\buildAgent\work
1>\7495521761d392b9\FodyIsolated\ModuleReader.cs:line 16\r
1>MSBUILD : error :    at InnerWeaver.Execute() in c:\TeamCity\buildAgent\work\74
1>95521761d392b9\FodyIsolated\InnerWeaver.cs:line 25\r
1>MSBUILD : error : Source:\r
1>MSBUILD : error : Mono.Cecil.Pdb\r
1>MSBUILD : error : TargetSite:\r
1>MSBUILD : error : Void ReadCustomMetadata(Microsoft.Cci.Pdb.BitAccess)\r
1>MSBUILD : error : 
1>  	Finished Fody 17ms.
========== Build: 0 succeeded, 1 failed, 7 up-to-date, 0 skipped ==========
```

**Comment by SimonCropp** [comment]
@bradphelan please try the newest version of fody

**Comment by jbevain** [comment]
Yes, that's fixed in master and 0.9.6.

**Comment by bradphelan** [comment]
Yeah. I found the problem that even if I was using the latest version of
fody but an older version was still in the packages folder it seemed to
fail. Works now.
